spacy/pipeline/pipes.pyx

```diff
@@ -1150,49 +1152,40 @@ class EntityLinker(Pipe):
             ents_by_offset = dict()
             for ent in doc.ents:
                 ents_by_offset[str(ent.start_char) + "_" + str(ent.end_char)] = ent
-            for entity in gold.links:
-                start, end, gold_kb = entity
+            for entity, kb_dict in gold.links.items():
+                start, end = entity
                 mention = doc.text[start:end]
+                for kb_id, value in kb_dict.items():
+                    entity_encoding = self.kb.get_vector(kb_id)
+                    prior_prob = self.kb.get_prior_prob(kb_id, mention)
+
+                    gold_ent = ents_by_offset[str(ent.start_char) + "_" + str(ent.end_char)]
+                    assert gold_ent is not None
```

Comment:
As @honnibal already mentioned, we always want to avoid `assert`s in non-test code, because they're not very user-friendly. So this should just raise a somewhat descriptive error `if gold_ent is None`.